container/kvm/initialisation.go

```diff
@@ -95,24 +89,36 @@ func getRequiredPackages(a string) []string {
 // createPool creates the libvirt storage pool directory. runCmd and chownFunc
 // are here for testing. runCmd so we can check the right shell out calls are
 // made, and chownFunc because we cannot chown unless we are root.
-func createPool(pathfinder func(string) (string, error), runCmd runFunc, chownFunc func(string) error) error {
+func ensurePool(poolInfo *libvirtPool, pathfinder func(string) (string, error), runCmd runFunc, chownFunc func(string) error) error {
 	poolDir, err := guestPath(pathfinder)
```

Comment:
If you do this at the top if this method:
```go
if poolInfo == nil {
    poolInfo = &libvirtPool{}
}
```

The if conditions below can be simplified to:
```go
if poolInfo.Name == "" {
...
if poolInfo.State != "running" {
...
if poolInfo.Autostart != "yes" {
...

Reply:
Done